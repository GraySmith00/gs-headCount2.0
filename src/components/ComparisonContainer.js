import React from 'react';
import PropTypes from 'prop-types';

import DistrictCard from './DistrictCard';
import '../css/ComparisonContainer.css';
import ComparisonCard from './ComparisonCard';

const ComparisonContainer = ({
  selectedDistricts,
  toggleSelected,
  category
}) => {
  let cardOne;
  let cardTwo;
  let comparisonCard;
  let hiddenCard;

  const selectedCards = selectedDistricts.length > 0;

  if (selectedDistricts[0]) {
    hiddenCard = (
      <div style={{ opacity: 0 }}>
        <DistrictCard
          {...selectedDistricts[0]}
          toggleSelected={toggleSelected}
          category={category}
          selectedCards={selectedCards}
        />
      </div>
    );
    cardOne = (
      <DistrictCard
        {...selectedDistricts[0]}
        toggleSelected={toggleSelected}
        category={category}
        selectedCards={selectedCards}
      />
    );
  }
  if (selectedDistricts[1]) {
    cardTwo = (
      <DistrictCard
        {...selectedDistricts[1]}
        toggleSelected={toggleSelected}
        category={category}
        selectedCards={selectedCards}
      />
    );
    comparisonCard = (
      <ComparisonCard
        selectedDistricts={selectedDistricts}
        category={category}
        selectedCards={selectedCards}
      />
    );
  }

  return (
    <div className="comparison-container">
      {hiddenCard}
      {cardOne}
      {comparisonCard}
      {cardTwo}
    </div>
  );
};

ComparisonContainer.propTypes = {
  selectedDistricts: PropTypes.array.isRequired,
  toggleSelected: PropTypes.func.isRequired,
  category: PropTypes.object.isRequired
};

export default ComparisonContainer;
