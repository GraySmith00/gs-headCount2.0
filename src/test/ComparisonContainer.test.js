import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import ComparisonContainer from '../components/ComparisonContainer';

describe('ComparisonContainer component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ComparisonContainer />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
