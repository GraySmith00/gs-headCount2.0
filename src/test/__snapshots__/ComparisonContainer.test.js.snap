// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer component should match the snapshot 1`] = `"<div class=\\"comparison-container\\"><div style=\\"opacity:0\\"><div class=\\"district-card\\" style=\\"width:9rem\\"><h3 class=\\"district-card-title\\">COLORADO</h3><div class=\\"district-stats\\"><p>2004 : <span class=\\"subPointFive\\">0.24</span></p><p>2005 : <span class=\\"subPointFive\\">0.278</span></p></div><p>AVG : <span class=\\"\\">0.53</span></p></div></div><div class=\\"district-card\\" style=\\"width:9rem\\"><h3 class=\\"district-card-title\\">COLORADO</h3><div class=\\"district-stats\\"><p>2004 : <span class=\\"subPointFive\\">0.24</span></p><p>2005 : <span class=\\"subPointFive\\">0.278</span></p></div><p>AVG : <span class=\\"\\">0.53</span></p></div><div class=\\"district-card\\"><h3 class=\\"district-card-title\\">COLORADO:</h3><p>0.53</p><h3>&lt;--- 0.768 ---&gt;</h3><h3 class=\\"district-card-title\\">ACADEMY 20:</h3><p>0.407</p></div><div class=\\"district-card\\" style=\\"width:9rem\\"><h3 class=\\"district-card-title\\">ACADEMY 20</h3><div class=\\"district-stats\\"><p>2010 : <span class=\\"\\">0.64</span></p><p>2011 : <span class=\\"\\">0.672</span></p></div><p>AVG : <span class=\\"subPointFive\\">0.407</span></p></div></div>"`;
